### ISSUE TITLE
`mypy.visitor` raises TypeError upon import in Python 3.6

### ISSUE THREAD

**Issue by speezepearson** [python/mypy#3679]
(Googling my error message in quotes turns up exactly zero results; I worry this will be irreproducible. Sorry if so!)

Symptom:
```
~ $ python3.6 -c 'import mypy.visitor'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/spencer/.local/lib/python3.6/site-packages/mypy/visitor.py", line 304, in <module>
    class NodeVisitor(Generic[T], ExpressionVisitor[T], StatementVisitor[T]):
  File "/usr/lib/python3.6/typing.py", line 944, in __new__
    self = super().__new__(cls, name, bases, namespace, _root=True)
  File "/usr/lib/python3.6/typing.py", line 118, in __new__
    return super().__new__(cls, name, bases, namespace)
  File "/usr/lib/python3.6/abc.py", line 133, in __new__
    cls = super().__new__(mcls, name, bases, namespace)
TypeError: Cannot create a consistent method resolution
order (MRO) for bases Generic, ExpressionVisitor, StatementVisitor
```

This is especially puzzling to me, because:
1. the MROs for those base classes haven't changed (which I can see by monkey-patching the `abc` module to print out the bases' MROs in `ABCMeta.__new__`), which suggests that 3.6 changed something about MRO-handling
2. the [release notes for Python 3.6](https://docs.python.org/3/whatsnew/3.6.html) don't mention any MRO-related change.

I've tried to recreate a similar inheritance structure without reference to MyPy, but haven't been able to get inconsistent behavior between 3.5 and 3.6.

Information about my installation:
```
~ $ which python3.6 pip3.6
/usr/bin/python3.6
/home/spencer/.local/bin/pip3.6
~ $ python3.6 --version
Python 3.6.0b2
~ $ pip3.6 show mypy
Name: mypy
Version: 0.511
Summary: Optional static typing for Python
Home-page: http://www.mypy-lang.org/
Author: Jukka Lehtosalo
Author-email: jukka.lehtosalo@iki.fi
License: MIT License
Location: /home/spencer/.local/lib/python3.6/site-packages
Requires: typed-ast
```

Is anybody able to reproduce this?

**Comment by gvanrossum** [comment]
We don't support using mypy as a package (it is only meant as an application). But I wonder if you should just upgrade your Python version -- you're using an old beta of 3.6.0 (3.6.0b2) and the version of typing.py in the stdlib might well be too ancient. Try installing Python 3.6.1.

**Comment by ilevkivskyi** [comment]
I am almost 100% sure this is because of an older ``typing`` from beta times. With older ``typing`` such errors sometimes occurred if ``Generic[T]`` is not the _last_ one in the base classes list.

**Comment by miedzinski** [comment]
I was experiencing this on Python 3.6.0-b2 from Ubuntu 16.10. Until 17.04 landed with 3.6.1 I was using self-compiled Python, so either upgrade your distro (if you're using Ubuntu derivative) or compile Python yourself.
